bobrun/wrap.go

```diff
@@ -0,0 +1,268 @@
+package bobrun
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/benchkram/bob/pkg/boblog"
+	"github.com/benchkram/bob/pkg/ctl"
+	"github.com/benchkram/errz"
+	"github.com/logrusorgru/aurora"
+	"mvdan.cc/sh/expand"
+	"mvdan.cc/sh/interp"
+	"mvdan.cc/sh/syntax"
+)
+
+// Wrapper wraps a run-task to provide init functionality executed after
+// the task started.
+type Wrapper struct {
```

Comment:
Maybe Wrapper is too general for name? I would change it to `InitWrapper` or `WithInit`.